src/services/session.service.ts

```diff
@@ -21,13 +23,75 @@ export class SessionService {
 
   async signUp(user: User) {
     try {
-      const registeredUser  = this.userService.createUser(user);
+      user.isVerified = false;
+      const registeredUser  = await this.userService.createUser(user);
+
+      this.sendVerificationToken({userId: registeredUser.id, email: registeredUser.email});
+
       return registeredUser;
     } catch (error) {
       throw new DatabaseError(ErrorsMessages.USER_ALREADY_EXISTS);
     }
   }
 
+  async sendVerificationToken(tokenData: AuthInterface.ITokenDataInput) {
+      const token = await this.jwtService.createJWT(tokenData);
+      this.redisService.addTokenToVerificationList({
+        email: tokenData.email,
+        token
+      });
+      
+      EmailService.sendEmail({
+        from: APPLICATION_EMAIL as string,
+        to: tokenData.email,
+        subject: 'Verify your MVD account',
+        text: token
+      });
+  }
+
+  async verifyUser(token: string) {
+    try {
+      //decode token and get email
+      const tokenDecoded = await this.jwtService.decodeJWT(token);
+      if (tokenDecoded === null) {

```

Comment:
One tip here, if you do just 

```typescript
if(!tokenDecoded){
   // your logic here
}
```

Is going to enter in the if condition if the variable is

- Or undefined
- Or null
- Or NaN
- Or '' -empty string-